go/types/leaf_sequence.go

```diff
@@ -91,10 +92,20 @@ func (seq leafSequence) writeTo(w nomsWriter) {
 }
 
 func (seq leafSequence) values() []Value {
+	return seq.valuesSlice(0, math.MaxUint64)
+}
+
+func (seq leafSequence) valuesSlice(from, to uint64) []Value {
 	dec, count := seq.decoderSkipToValues()
-	vs := make([]Value, count)
-	for i := uint64(0); i < count; i++ {
-		vs[i] = dec.readValue()
+	if to > count {
+		to = count
+	}
+	for i := uint64(0); i < from; i++ {
```

Comment:
There is a decoderSkipToIndex already that uses the offset to jump to the right spot without seeking